src/docker_git.nim

```diff
@@ -54,8 +54,10 @@ proc createTempKnownHosts(data: string): string =
   if data == "":
     return ""
   let (f, path) = getNewTempFile()
-  f.write(data)
-  f.close()
+  try:
```

Comment:
See above, should have a 'writeNewTempFile()`; Plus, it occurs to me we should have a semgrep rule to warn on new adds of getNewTempFile()

Reply:
cc @indecisivedragon was just mentioning semgrep to me as well. good idea to add that to chalk CI